components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/IdentityProviderManagementServiceTest.java

```diff
@@ -0,0 +1,904 @@
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.IdentityProviderProperty;
+import org.wso2.carbon.identity.application.common.model.LocalRole;
+import org.wso2.carbon.identity.application.common.model.PermissionsAndRoleConfig;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.model.ProvisioningConnectorConfig;
+import org.wso2.carbon.identity.application.common.model.RoleMapping;
+import org.wso2.carbon.identity.claim.metadata.mgt.ClaimMetadataManagementServiceImpl;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.LocalClaim;
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.common.testng.WithRealmService;
+import org.wso2.carbon.identity.common.testng.WithRegistry;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.idp.mgt.internal.IdpMgtServiceComponentHolder;
+import org.wso2.carbon.idp.mgt.util.MetadataConverter;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertThrows;
+import static org.wso2.carbon.base.MultitenantConstants.SUPER_TENANT_ID;
+
+
+/**
+ * Unit test cases for IdentityProviderManagementService.
+ */
+@Test
+@WithCarbonHome
+@WithRegistry
+@WithRealmService (injectToSingletons = {IdpMgtServiceComponentHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = { "dbscripts/h2.sql" })
+
+public class IdentityProviderManagementServiceTest  extends PowerMockTestCase {
+
+    @Mock
+    MetadataConverter mockMetadataConverter;
+
+    @BeforeMethod
+    public void init() throws Exception {
+        List<MetadataConverter> list = Arrays.asList(mockMetadataConverter);
+        IdpMgtServiceComponentHolder.getInstance().setMetadataConverters(list);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+        IdentityProviderManagementService identityProviderManagementService = new IdentityProviderManagementService();
+        //Clear Database after every test
+        removeTestIdps(identityProviderManagementService);
+    }
+
+
+    @DataProvider
+    public Object[][] testAddIdPData() {
```

Comment:
'test' prefix should only be there for tests. refrain from using that prefix for data providers

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3493/commits/09fbd1468a2abedaaf4c5f79daa567709fcbda99